amundsen_application/static/js/fixtures/mockRouter.ts

```diff
@@ -0,0 +1,40 @@
+import { RouteComponentProps } from 'react-router';
+
+import * as History from 'history';
+
+// Mock React-Router
+export function getMockRouterProps<P>(data: P, location: History.Location): RouteComponentProps<P> {
+  const defaultLocation: History.Location = {
+    hash: '',
+    key: '',
+    pathname: '',
+    search: '',
+    state: {},
+  };
+  const props: RouteComponentProps<P> = {
+    match: {
+      isExact: true,
+      params: data,
+      path: '',
+      url: '',
+    },
+    location: location || defaultLocation,
+    history: {
+      length: 2,
+      action: 'POP',
+      location: location || defaultLocation,
+      push: () => {},
+      replace: null,
+      go: null,
+      goBack: null,
+      goForward: null,
+      block: null,
+      createHref: null,
+      listen: null,
+    },
+    staticContext: {
+    }
```

Comment:
I see ` staticContext: jest.fn() as any` elsewhere in code, but that might be wrong. Looking at the type info it is 

```
extends StaticContext {
  statusCode?: number;
}
```

Reply:
Yeah the `jest.fn()` is incorrect. It's an object not a method, and I left it empty because I don't quite know what it does. 

Notice most things that we don't use I've set as sort of the bare minimum needed to pass the typechecker, because I'm concerned about adding in anything that would unknowingly have side effects. The way I currently see it, we add more meat to this only when that value is relevant for our tests.